psi4/src/psi4/libqt/CMakeLists.txt

```diff
@@ -42,9 +52,11 @@ endif()
 psi4_add_module(lib qt sources)
 target_compile_definitions(qt
   PRIVATE
+    $<$<BOOL:${_has_daxpby}>:BLAS_HAS_DAXPBY>
```

Comment:
I do not think this will work as well as you would hope for. When working on #2823 I have initially tried to write a fancy build-system check that would have errored out early and told the user if they lack DGGSVD3/DGGSVP3, but when I tried it I realized that detection is broken, so I abandoned the entire concept of giving an error message early.

It has been a while since I was working on it, but I _think_ that the problem was that for some reason CMake was always failing to find DGGSVD3/DGGSVP3, even though it was available, so the new non-deprecated functions were not getting used, ever.

I have abandoned the idea of completely debugging CMake detection of LAPACK extensions, but I seem to recall that one of the issues was that `check_fortran_function_exists` does not (and probably cannot) get called if `Fortran_ENABLED == false`. So having these extension availability checks work reliably may come at the cost of always requiring a Fortran compiler, or something like that.

Reply:
Seeking this simple check also gave me fits this weekend. I'm pretty sure the PR can detect both found and missing w/o Fortran for at least the Linux+MKL case. MKL has symbols for 3 of the 4 FC_MANGLE variations, and I don't hear much about people having to change it, so I'm hoping that =2 is fairly settled. I'd be content if this check works only for MKL or Accelerate and lets one avoid the Fortran compiler.

I'd much prefer `check_cxx_source_runs`, and I have a snippet, but (1) it's calling `cblas_daxpby` and (2) it has `mkl.h` explicitly which won't do for multiple backends. So I share your reservations, but I'd like to see an Accelerate test to see if it's any use.

Reply:
I actually would not mind at all if having a Fortran compiler would become an essential build-time dependency. In fact, requiring a modern (F2003 or newer) compiler would actually open the way for completely eradicating having to deal with FC_MANGLE at all. (if someone finds the time to write the required shims)

I would strongly prefer if Psi4 retained at least some compatibility with BLAS/LAPACK implementations beyond MKL/Accelerate, like OpenBLAS which does implement extensions like daxpby.
It is disappointing that Apple's budget does not seem to cover updating their LAPACK library with contemporary extensions.